sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -257,6 +257,7 @@ trait Row extends Serializable {
    *
    * @throws ClassCastException when data type does not match.
    */
+  // TODO(davies): This is not the right default implementation, we use Int as Date internally
```

Comment:
Are we excepting a `Date` object or its `Int` representation? For internal use, I guess we expect an int. But, since users can also call it, for them, a `Date` object is expected, right?
Right now, if I call `getDate`, I will get a `ClassCastException`?

Can you file a jira for it for 1.4 and mark it as a blocker?